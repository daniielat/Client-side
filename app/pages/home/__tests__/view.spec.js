const React = require('react');
const HomeView = require('../view');
const { render, screen, act, waitFor } = require('@testing-library/react');
require('core-js');
const restClient = require('nordic/restclient');
jest.mock('nordic/restclient');

const { mockGet } = restClient;

describe('La view de Home', () => {
    let component; 

    beforeEach(async () => {  
        await act(async () => {
            mockGet.mockResolvedValueOnce({data: [{id: 1, title: 'Samsung'}]});
            component = await waitFor(() => render(<HomeView />));
        });
    });
    
    it('1) Renderiza', async () => {
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();
    });

    it('2) Utiliza restclient desde la view', async () => {
        expect(mockGet).toHaveBeenCalled();
    });
        
    it('3) Renderiza los productos de la API dentro de una lista', async () => {
        await act(async () => {
            const products = await screen.findAllByRole('listitem');
            expect(products).toHaveLength(1);
        })
    });
});