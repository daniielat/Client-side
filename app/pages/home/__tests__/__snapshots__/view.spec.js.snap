// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`La view de Home 1) Renderiza 1`] = `
<DocumentFragment>
  <div
    class="demo"
  >
    <ul>
      <li>
        <h2>
          Samsung
        </h2>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
